packages/common/src/project/models.ts

```diff
@@ -195,4 +196,7 @@ export class CustomDataSourceBase<
   @IsOptional()
   @IsString()
   abi: string;
+  @IsOptional()
+  @IsEthereumAddress()
+  address: string;
```

Comment:
@ianhe8x would it make sense to nest this under a `filter` key? Or maybe change abi to `abi: { name: 'erc20'; address: '0x...'}`

Reply:
I would say parallel to abi，so now is fine. Depends on where is is used, if you add a extra filtering step for as, then under filter looks better